linux/src/file_linux.c

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "file.h"
+#include "file_internal.h"
+
+MU_DEFINE_ENUM_STRINGS(FILE_ASYNC_OPERATION, FILE_ASYNC_OPERATION_VALUES)
+
+typedef struct FILE_HANDLE_DATA_TAG
+{
+    int handle;
+    FILE_REPORT_FAULT user_report_fault_callback;
+    void* user_report_fault_context;
+}FILE_HANDLE_DATA;
+
+typedef struct FILE_LINUX_WRITE_TAG
+{
+    FILE_HANDLE_DATA handle;
+    FILE_WRITE_DATA_CONTEXT write_data;

```

Comment:
>FILE_WRITE_DATA_CONTEXT [](start = 4, length = 23)

Consider not having this structure (and file_internal.h can go away) and hold in FILE_LINUX_WRITE everything needed.

Reply:
Yeah that makes sense. Coming up in next iteration.

---
In reply to: [452464984](https://github.com/Azure/azure-c-pal/pull/7#discussion_r452464984) [](ancestors = 452464984)